### ISSUE TITLE
Failed to install SingleR

### ISSUE THREAD

**Issue by zxcary** [LTLA/SingleR#152]
I used the following code to install this package, but it failed.
`BiocManager::install("LTLA/SingleR")`
`fine_tune_de.cpp:1:10: fatal error: beachmat3/beachmat.h: No such file or directory
 #include "beachmat3/beachmat.h"
          ^~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [C:/PROGRA~1/R/R-40~1.2/etc/x64/Makeconf:229: fine_tune_de.o] Error 1
ERROR: compilation failed for package 'SingleR'
* removing 'C:/Users/zxc/Documents/R/win-library/4.0/SingleR'
ERROR: Failed to install 'SingleR' from GitHub:
  (converted from warning)installation of package ‘C:/Users/zxc/AppData/Local/Temp/RtmpW2GFW9/file4f6c431f1e1c/SingleR_1.3.11.tar.gz’ had non-zero exit status`

**Comment by j-andrews7** [comment]
Need the `sessionInfo()`.

**Comment by dtm2451** [comment]
Yes, please give your `sessionInfo()`.  Also placing ` ``` ` above and below your code chunks / error messages will will them be formatted in a more readable way.

It's likely that you are getting this error because you have the release versions of Bioconductor package dependencies, whereas the github syncs up with the development version of SingleR and requires the development versions of its Bioc dependencies.

**Comment by 1874Mercury** [comment]
I got the same problem, 
```
fine_tune_de.cpp:1:10: fatal error: beachmat3/beachmat.h: No such file or directory
 #include "beachmat3/beachmat.h"
          ^~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [/Share2/home/zhaolab/miniconda3/envs/R4.0/lib/R/etc/Makeconf:176: fine_tune_de.o] Error 1
ERROR: compilation failed for package ‘SingleR’
* removing ‘/Share2/home/zhaolab/miniconda3/envs/R4.0/lib/R/library/SingleR’
Error: Failed to install 'SingleR' from GitHub:
  (converted from warning) installation of package ‘/tmp/RtmpIKmys4/file1b373228059e7/SingleR_1.3.11.tar.gz’ had non-zero exit status
```

and this is my session info:

```
> sessionInfo()
R version 3.6.3 (2020-02-29)
Platform: x86_64-conda_cos6-linux-gnu (64-bit)
Running under: Red Hat Enterprise Linux Server release 6.6 (Santiago)

Matrix products: default
BLAS/LAPACK: /Share2/home/zhaolab/miniconda3/envs/R4.0/lib/libopenblasp-r0.3.10.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_3.6.3
```

**Comment by dtm2451** [comment]
@1874Mercury, looks like you're using R-3.6.  However, the github version of SingleR targetted by `BiocManager("LTLA/SingleR")` requires dependencies that are only tested and compiled for R-4.0.

My guess would be that the versions of dependencies downloaded and installed when you are trying to install SingleR, are the latest ones released for your version of R.  Thus they are not up to date, and thus the missing needed file.

I see three options then, but I don't recommend the 3rd:
1. Use the version of SingleR released on R-3.6, v1.0, which you should be able to get via `BiocManager::install("SingleR")`.
2. Upgrade your R to 4.0, then try again.
3. Stick with R-3.6, but find, and install from source, the current versions of all of SingleR's dependencies. Then try installing SingleR again.  But... since the neither SingleR nor most of its dependencies are built/tested for R-3.6, there is no guarantee that such a method would be successful && it seems like a lot of work for potentially nothing.

**Comment by 1874Mercury** [comment]
It worked!

Thanks a ton :)
